crates/aptos-rest-client/src/lib.rs

```diff
@@ -1006,6 +1012,58 @@ impl Client {
         let (response, state) = self.check_response(response).await?;
         Ok(Response::new(response.bytes().await?, state))
     }
+
+    pub async fn try_until_ok<F, Fut, T>(
```

Comment:
We should consider leveraging a library that already does this instead of writing the umpteenth backoff function in our code (@perryjrandall just managed to get rid of one). `again` seems to work pretty well.